winml/lib/Api.Image/DeviceHelpers.cpp

```diff
@@ -120,6 +120,19 @@ HRESULT _winml::GetDXCoreHardwareAdapterWithPreference(DXGI_GPU_PREFERENCE prefe
   firstHardwareAdapter.copy_to(ppAdapter);
   return S_OK;
 }
+
+HRESULT _winml::GetDXCoreHardwareAdapterForVPU(_COM_Outptr_ IDXCoreAdapter** ppAdapter) {
+  winrt::com_ptr<IDXCoreAdapterFactory> spFactory;
+  winrt::com_ptr<IDXCoreAdapterList> spAdapterList;
+  const GUID gpuFilter[] = {DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE};
+  RETURN_IF_FAILED(spFactory->CreateAdapterList(1, gpuFilter, IID_PPV_ARGS(spAdapterList.put())));
+  winrt::com_ptr<IDXCoreAdapter> spCurrAdapter;
+  RETURN_IF_FAILED(spAdapterList->GetAdapter(0, IID_PPV_ARGS(spCurrAdapter.put())));
+  bool isHardware;
```

Comment:
This is perhaps more of a question of API design, but what happens if adapter 0 is actually a GPU? MCDM and 1_0_CORE devices are (strictly speaking) subsets of fully-fledged GPUs, so I think it's valid for regular GPUs to show up even when filtering by `DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE` (because they satisfy all the constraints of a CORE_COMPUTE device).

Is the intent of the `LearningModelDeviceKind.ComputeAccelerator` to allow WinML to select from the union of GPU+VPU devices, or is the intent for it to exclusively target VPUs only?

Reply:
@smk2007 as this has has API consequences for WinML